src/pages/_app.tsx

```diff
@@ -1,20 +1,37 @@
 import { Analytics } from "@vercel/analytics/react"
-import { AppProps } from "next/app"
+import App, { AppContext, AppProps } from "next/app"
+import { ReactNode } from "react"
 
 import "@/styles/globals.css"
 
 import { inter, vt323 } from "@/lib/fonts"
 
-function MyApp({ Component, pageProps }: AppProps) {
+import AppProviders from "@/components/AppProviders"
+
+type FortressAppProps = Pick<AppProps, "Component" | "pageProps"> & {
+  isApp: boolean
+}
+
+function FortressApp({ Component, pageProps, isApp }: FortressAppProps) {
+  const Providers = isApp
```

Comment:
I originally put `<AppProviders />` in `Layout.tsx` with the intention of not having providers loaded on the homepage. This has proven to be annoying because you can't use anything that relies on the providers in a root page file.

This solution checks the pathname and includes the providers one level higher than `Layout` for any path that isn't the landing page.